### ISSUE TITLE
PDF: Broken reference to built-in exceptions in tutorial

### ISSUE THREAD

**Issue by shekelboi** [python/cpython#115698]
# Documentation

The reference to the built-in exceptions is broken in the tutorial
![image](https://github.com/python/cpython/assets/38100924/aa6f981c-3791-4a4d-b7f8-248a2d531cd8)
![image](https://github.com/python/cpython/assets/38100924/ac499a49-4445-4693-8296-df546cd65b98)

It is worth noting that this only affects the PDF version of the tutorial.

**Comment by willingc** [comment]
@hugovk I'm confirming that this behavior is happening in the PDF and not the html version.

It's possible that the `:ref:` links to a different PDF than the tutorial PDF and the sphinx pdf builder that we use does not support linking to a different PDF.

**Comment by CAM-Gerlach** [comment]
> It's possible that the :ref: links to a different PDF than the tutorial PDF and the sphinx pdf builder that we use does not support linking to a different PDF.

Yup, that appears to be the case for all xref roles; they work intra-PDF but not cross-PDF. However, it seems that's likely less a bug in Sphinx and more a fundamental limitation the fundamental limitations of the format involved, as I'm not aware of a standardized mechanism in the PDF format to tell the reader to open another specific anchor in another PDF file at a relative location to the current one (and even then, that would still break if the PDFs were compressed, moved, renamed ,etc).

What does seem possible, though is for Sphinx to at least insert the title of the linked heading instead of just the raw ref target label text (since it _can_ resolve the xref at build time, it just can't make it a link in the PDF builder) which would be more useful to the reader and look less like a bug.

**Comment by willingc** [comment]
Ideally, I would love to see us move toward something else to generate the PDF other than LaTeX in the future. I keep hoping that someone with time would write a converter from html to PDF for Sphinx that uses Playwright to do the conversion.

**Comment by CAM-Gerlach** [comment]
> @CAM-Gerlach You just jogged my memory. I think at some point with Jupyter that we were running the latex command twice to get the references to resolve.

_Intra-file_ references appear to working fine now (I had dug up some similar-sounding issues in the LaTeX PDF builder in the past looking into this issue, but as far as I can tell they've mostly been resolved). However, per my main point above, I'm not sure the PDF format _has_ a standard mechanism for linking to an _entirely separate file_ on an arbitrary local machine via a relative path, and even if it did, it would be rather fragile as discussed. I'm not sure how using a different PDF builder would solve what appears to be fundamental limitation with the format, but maybe there's something I'm missing here?

**Comment by willingc** [comment]
@CAM-Gerlach I believe that having LaTeX as an intermediary to generate the PDF versus going directly from HTML to PDF. This is definitely something that we can put on the backburner until PyCon.
